### ISSUE TITLE
The elasticsearch/elasticsearch package is required for Elasticsearch support after installation

### ISSUE THREAD

**Issue by lkaopremier** [api-platform/core#5009]
**API Platform version(s) affected**: 3.0

**Description**  
After adding the elasticsearch/elasticsearch package, I got this error "The elasticsearch/elasticsearch package is required for Elasticsearch support.", When analyzing the source code of the version of elasticsearch I added I realized that the namespace declared in the elasticsearch client is different from the namespace used by Api Platform.

**How to reproduce**  
Using version 3.0 of the Api platform and adding the latest version of the elasticsearch/elasticsearch package finally enabled elasticsearch in the configs, [See the official doc](https://api-platform.com/docs/v2.5/core/elasticsearch/#elasticsearch-support).

**Additional Context**  
![Capture d’écran 2022-09-24 à 11 25 23](https://user-images.githubusercontent.com/36473778/192095441-0c5a3355-b8d4-4565-bb72-ed4d11dad913.png)
![Capture d’écran 2022-09-24 à 11 30 33](https://user-images.githubusercontent.com/36473778/192096256-70c56122-a1e5-4f5a-a70b-5a7edd3b2ebf.png)
![Capture d’écran 2022-09-24 à 11 43 01](https://user-images.githubusercontent.com/36473778/192096259-451cafb3-b740-41c3-8104-ae9e98c4eee6.png)

**Comment by darthf1** [comment]
Hi @alanpoulain,

Would you be able to give some pointers, on how to properly support both ES 7 and ES 8? I was planning to migrate to ES 8 in a few weeks so depending on the work I could give it a go at that time.

Current usages of the `Elasticsearch` namespace (in 2.6.8):
![image](https://user-images.githubusercontent.com/17253332/192725562-aed2aa9f-ff0d-4707-8cf6-5b0d433251d6.png)

**Comment by alanpoulain** [comment]
You probably need to use a facade to abstract the client used. You have an example here: https://github.com/api-platform/core/blob/2.7/src/Util/Inflector.php

**Comment by curry684** [comment]
I just upgraded `api-platform/core` to `^3.0` in my composer.json, and accidentally happened to notice that afterwards I got stuck on 3.0.0 without upgrade possibility to 3.0.2. Further investigation got me to this issue, which is all fine and dandy, except that I'm now on 3.0.0, with Elasticsearch 8 deeply embedded in my project. So I should remain stuck on 2.7 for the time being, not 3.0.0, the last version without this conflict.

I would dearly recommend yanking 3.0.0 to prevent this confusion which could've caused me severe headaches had I not run `composer outdated` directly after upgrading.
